src/libmongoc/tests/test-mongoc-http.c

```diff
@@ -34,24 +34,48 @@ test_mongoc_http (void *unused)
 
    /* Basic GET request */
    req.method = "GET";
-   req.host = "example.com";
+   req.host = "httpbin.org";
+   req.path = "get";
    req.port = 80;
-   // Empty body is okay
-   req.body = "";
-   req.body_len = 0;
    r = _mongoc_http_send (&req, 10000, false, NULL, &res, &error);
-   ASSERT_CMPINT (res.status, ==, 200);
+   ASSERT_WITH_MSG (res.status == 200,
+                    "unexpected status code %d\n"
+                    "RESPONSE BODY BEGIN\n"
+                    "%s"
+                    "RESPONSE BODY END\n",
+                    res.status,
+                    res.body_len > 0 ? res.body : "");
    ASSERT_OR_PRINT (r, error);
```

Comment:
```suggestion
   ASSERT_OR_PRINT (r, error);
   ASSERT_WITH_MSG (res.status == 200,
                    "unexpected status code %d\n"
                    "RESPONSE BODY BEGIN\n"
                    "%s"
                    "RESPONSE BODY END\n",
                    res.status,
                    res.body_len > 0 ? res.body : "");
```

Suggest checking for an error first. An error in socket connection may result in `res.status` being `0`. The `error` may have more useful information.